plugins/tickets.py

```diff
@@ -0,0 +1,1821 @@
+    """
+    if not tickets:
+        return None
+
+    lines = [ticket.summary(fmt=fmt) for ticket in tickets]
+    blocks = ['\n'.join(lines[i:i+10]) for i in range(0, len(lines), 10)]
+    page_count = len(blocks)
+
+    embeds = (discord.Embed(description=blocks[i], title=title) for i in range(page_count))
+
+    if page_count > 1:
+        embeds = (embed.set_footer(text="Page {}/{}".format(i+1, page_count)) for i, embed in enumerate(embeds))
+
+    return embeds
+
+
+Page = namedtuple('Page', ('content', 'embed'), defaults=(None, None))
+
+
+async def pager(dest: discord.abc.Messageable, *pages):
+    """
+    Page a sequence of pages.
+    """
+    _next_reaction = '⏭️'
+    _prev_reaction = '⏮️'
+
+    # Sanity check
+    if not pages:
+        raise ValueError("Cannot page with no pages!")
+
+    # Send first page
+    msg = await dest.send(**pages[0]._asdict())
+
+    if len(pages) == 1:
+        return
+
+    # Add reactions
+    await msg.add_reaction(_prev_reaction)
+    await msg.add_reaction(_next_reaction)
+
+    index = 0
+    while True:
+        try:
+            payload = await discord_client.client.wait_for(
+                'raw_reaction_add', timeout=120,
+                check=lambda p: (p.message_id == msg.id
+                                 and str(p.emoji) in [_next_reaction, _prev_reaction]
+                                 and p.user_id != msg.guild.me.id)
+            )
+            if str(payload.emoji) == _next_reaction:
+                index += 1
+            elif str(payload.emoji) == _prev_reaction:
+                index -= 1
+            index %= len(pages)
+            await msg.edit(**pages[index]._asdict())
+            try:
+                await msg.remove_reaction(payload.emoji, discord.Object(payload.user_id))
+            except discord.HTTPException:
+                pass
+        except asyncio.TimeoutError:
+            break
+        except asyncio.CancelledError:
+            break
+
+
+@commands.command("note")
+@priv.priv("mod")
+async def cmd_note(msg: discord.Message, args):
+    """
+    Create a note on the target user.
+    """
+    if not isinstance(target_arg := args.next_arg(), commands.UserMentionArg):
+        # TODO: Usage
+        return
+    targetid = target_arg.id
+
+    maybe_note_arg = args.next_arg()
+    note = None
+    if maybe_note_arg is None:
+        # Request the note dynamically
+        await msg.channel.send("Please enter the note contents, or send `c` to cancel:")
+        try:
+            message = await discord_client.client.wait_for(
+                'message',
+                timeout=300,
+                check=lambda msg_: (msg_.channel == msg.channel) and (msg_.author == msg.author)
+            )
+        except asyncio.TimeoutError:
+            await msg.channel.send("Note prompt timed out, please try again.")
+        if message.content.lower() == 'c':
+            await msg.channel.send("Note prompt cancelled, no note was created.")
```

Comment:
Not a fan of this `c` thing. Could we use a :x: reaction instead?
P.S. `asyncio.wait(return_when=FIRST_COMPLETED)`